edge-util/src/Microsoft.Azure.Devices.Edge.Util/Option.cs

```diff
@@ -127,6 +127,19 @@ public T Expect<TException>(Func<TException> exception)
         [Pure]
         public TResult Match<TResult>(Func<T, TResult> some, Func<TResult> none) => this.HasValue ? some(this.Value) : none();
 
+        [Pure]
+        public void MatchNoReturn(Action<T> some, Action none)
```

Comment:
I wish this was more like Rust, and we could just do `option.map(some).or_else(none)`.
Would this alternate make sense to people?
```
        public void ForEach(Action<T> action, Action none = null)
        {
            if (this.HasValue)
            {
                action(this.Value);
            }
            else
            {
                none?.Invoke();
            }
        }
```

Reply:
`ForEachAsync`  could be:
```
public Task ForEachAsync(Func<T, Task> action, Func<Task> none = null) => this.HasValue ? action(this.Value) : none?.Invoke();
```

Reply:
I also like adding an overload to ForEach better...

Reply:
Updated to overload ForEach